### PR TITLE
Add `MappingDstNsRemover`

### PR DESCRIPTION
**Pull request by NebelNidas**
To remove unwanted destination namespaces.

### CODE REVIEW

File: src/main/java/net/fabricmc/mappingio/adapter/MappingDstNsRemover.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import net.fabricmc.mappingio.MappedElementKind;
+import net.fabricmc.mappingio.MappingVisitor;
+
+public final class MappingDstNsRemover extends ForwardingMappingVisitor {
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/mappingio/adapter/MappingDstNsRemover.java:29]: Add some docs

File: src/main/java/net/fabricmc/mappingio/adapter/MappingDstNsRemover.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	public MappingDstNsRemover(MappingVisitor next, Integer... namespacesToRemove) {
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/mappingio/adapter/MappingDstNsRemover.java:30]: Remove this constructor, its not useful when using the visitor pattern. Only take string namespaces.

File: src/main/java/net/fabricmc/mappingio/adapter/MappingDstNsRemover.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		super(next);
+
+		Objects.requireNonNull(namespacesToRemove, "null namespacesToRemove array");
+
+		this.namespacesIndicesToRemove = Arrays.asList(namespacesToRemove);
+		this.namespaceNamesToRemove = Collections.emptyList();
+	}
+
+	public MappingDstNsRemover(MappingVisitor next, String... namespacesToRemove) {
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/mappingio/adapter/MappingDstNsRemover.java:39]: Id just take a `List<String>` makes it easier to add more constructors later. Would also be similar to MappingDstNsReorder

**Review by modmuss50 — CHANGES_REQUESTED** [review]
Would be nice to start adding some unit tests for these, but thats likely out of scope for this PR.

**Comment by NebelNidas** [comment]
`MappingDstNsReorder` seems to be able to do the same by omitting the undesired namespace in the `newDstNs` list

**Pull request closed without merging**